e2e/runner/sui.go

```diff
@@ -1,17 +1,100 @@
 package runner
 
 import (
+	"encoding/hex"
 	"fmt"
+	"math/big"
+	"strconv"
 	"strings"
 
 	"cosmossdk.io/math"
 	"github.com/block-vision/sui-go-sdk/models"
 	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
 	"github.com/stretchr/testify/require"
+	"github.com/zeta-chain/protocol-contracts/pkg/gatewayzevm.sol"
 
 	"github.com/zeta-chain/node/pkg/contracts/sui"
 )
 
+// SuiGetSUIBalance returns the SUI balance of an address
+func (r *E2ERunner) SuiGetSUIBalance(addr string) uint64 {
+	resp, err := r.Clients.Sui.SuiXGetBalance(r.Ctx, models.SuiXGetBalanceRequest{
+		Owner:    addr,
+		CoinType: string(sui.SUI),
+	})
+	require.NoError(r, err)
+
+	balance, err := strconv.ParseUint(resp.TotalBalance, 10, 64)
+	require.NoError(r, err)
+
+	return balance
+}
+
+// SuiGetFungibleTokenBalance returns the fungible token balance of an address
+func (r *E2ERunner) SuiGetFungibleTokenBalance(addr string) uint64 {
```

Comment:
Let's make it more streamlined; currently imo it's not clear that is would read asset type from the runner.
```suggestion
func (r *E2ERunner) SuiGetFungibleTokenBalance(assetType, account string) uint64 {
```

Reply:
All functions in runner referencing FungibleToken are using the `SuiTokenZRC20` set from the config, this is same as the ERC20 or the SPL set, this seems to be unnecessary added complexity to me as there is likely for the future no other asset you would use with this function